src/blk/kernel/KernelDevice.cc

```diff
@@ -587,13 +587,35 @@ bool KernelDevice::_discard_started()
   return !discard_threads.empty();
 }
 
-void KernelDevice::discard_drain()
+int KernelDevice::discard_drain(uint32_t timeout_msec = 0)
 {
   dout(10) << __func__ << dendl;
+  bool check_timeout = false;
```

Comment:
another option: don't reuse timeout_msec, create a new msec variable, and use the original instead of creating the 'check_timeout' boolean.